src/libs/ReportUtils.ts

```diff
@@ -2201,6 +2201,34 @@ function getModifiedExpenseOriginalMessage(oldTransaction: OnyxEntry<Transaction
     return originalMessage;
 }
 
+/**
+ * Build invited usernames for admin chat threads
+ * @param parentReportAction
+ */
+function getAdminRoomInvitedParticipants(parentReportAction: ReportAction | Record<string, never>) {
+    const originalMessage = parentReportAction?.originalMessage as ChangeLog | undefined;
+    let actionMessage: string;
+    const verb =
+        parentReportAction.actionName === CONST.REPORT.ACTIONS.TYPE.ROOMCHANGELOG.INVITE_TO_ROOM || parentReportAction.actionName === CONST.REPORT.ACTIONS.TYPE.POLICYCHANGELOG.INVITE_TO_ROOM
+            ? Localize.translateLocal('workspace.invite.invited')
+            : Localize.translateLocal('workspace.invite.removed');
+    const participantAccountIDs = originalMessage?.targetAccountIDs ?? [];
+    const participants = participantAccountIDs.map((id) => getDisplayNameForParticipant(id));
+    const users = participants.length > 1 ? participants.join(` ${Localize.translateLocal('common.and')} `) : participants[0];
+    actionMessage = `${verb} ${users}`;
+
+    const roomName = originalMessage?.roomName ?? '';
+    if (roomName) {
+        const preposition =
+            parentReportAction.actionName === CONST.REPORT.ACTIONS.TYPE.ROOMCHANGELOG.INVITE_TO_ROOM ||
+            parentReportAction.actionName === CONST.REPORT.ACTIONS.TYPE.POLICYCHANGELOG.INVITE_TO_ROOM
+                ? ` ${Localize.translateLocal('workspace.invite.to')}`
+                : ` ${Localize.translateLocal('workspace.invite.from')}`;
+        actionMessage += `${preposition} ${roomName}`;
+    }
+    return actionMessage;
+}
```

Comment:
Great work here!! I see some opportunity to increase readability and performance a tad bit. Here's my suggestion:
1. Early return if `parentReportAction` or `parentReportAction.originalMessage` is not present. We can the skip the whole computation whatsoever and also simplify some typings and if statements later.
2. The check for the action type (INVITE_TO_ROOM) to determine the verb and preposition is performed only once. This reduces redundancy and improves performance slightly.
3. Streamlining the construction of `actionMessage` using a template literal which avoids using the if statement and makes things more readable.
4. Mapping the `participantsAccountIDs` array doesn't require creating an ad hoc unnamed arrow function. Just pass `getDisplayNameForParticipant` as a mapper.

```suggestion
function getAdminRoomInvitedParticipants(parentReportAction: ReportAction | Record<string, never>) {
    if (!parentReportAction || !parentReportAction.originalMessage) {
        return '';
    }

    const originalMessage = parentReportAction.originalMessage as ChangeLog;
    const actionType = parentReportAction.actionName;
    const isInviteAction = actionType === CONST.REPORT.ACTIONS.TYPE.ROOMCHANGELOG.INVITE_TO_ROOM ||
                           actionType === CONST.REPORT.ACTIONS.TYPE.POLICYCHANGELOG.INVITE_TO_ROOM;

    const verbKey = isInviteAction ? 'workspace.invite.invited' : 'workspace.invite.removed';
    const verb = Localize.translateLocal(verbKey);

    const participantAccountIDs = originalMessage.targetAccountIDs ?? [];
    const participants = participantAccountIDs.map(getDisplayNameForParticipant);
    const users = participants.length > 1 ? participants.join(` ${Localize.translateLocal('common.and')} `) : participants[0];

    const prepositionKey = isInviteAction ? 'workspace.invite.to' : 'workspace.invite.from';
    const preposition = ` ${Localize.translateLocal(prepositionKey)}`;
    const roomName = originalMessage.roomName ?? '';

    const actionMessage = roomName 
        ? `${verb} ${users}${preposition} ${roomName}`
        : `${verb} ${users}`;

    return actionMessage;
}
```

Reply:
@lukemorawski thanks for feedback!
About point 4 - i was trying to do that - but it will break TS check because getDisplayNameForParticipant is expecting some other parameters - in case that `map` function has 3 other params` (value: T, index: number, array: T[]) => U, thisArg?: any) - which is not equal to getDisplayNameForParticipant function
`
For all other points i agree - and will adapt readability!